packages/rn-tester/js/examples/Animated/ColorStylesExample.js

```diff
@@ -42,7 +44,8 @@ function AnimatedView({useNativeDriver}: {useNativeDriver: boolean}) {
   const interpolationAnimatedStyle = {
     backgroundColor: animatedBaseValue.interpolate({
       inputRange: [0, 1],
-      outputRange: ['blue', 'red'],
+      // outputRange: ['blue', 'red'],
+      outputRange: ['color(display-p3 0 0 1)', 'color(display-p3 1 0 0)'],
```

Comment:
@cipolleschi I have interpolate working without the native driver but if useNativeDriver is true the output ends in black after interpolating from blue to red. I'm a bit confused because AFAICT the output range is [derived from the UIColor objects](https://github.com/facebook/react-native/blob/782e9eace974baefba7d8978739f11fe1ea70dc9/packages/react-native/Libraries/NativeAnimation/RCTAnimationUtils.mm#L87) that we get from [RCTConvert](https://github.com/facebook/react-native/blob/782e9eace974baefba7d8978739f11fe1ea70dc9/packages/react-native/Libraries/NativeAnimation/Nodes/RCTInterpolationAnimatedNode.mm#L108). I was thinking with the extended gamut that we might be getting values greater than 1 for the maximums there but when logging that doesn't appear to be the case. Any insights here?

Reply:
Actually, yes, that's the case. Not only greater than 1, but even negative.

For example, if you try this code:

```objc
UIColor *c = [UIColor colorWithDisplayP3Red:1 green:0.75 blue:0.25 alpha:1];
CGFloat red, green, blue, alpha;
[c getRed:&red green:&green blue:&blue alpha:&alpha];
```
And then you debug it, you'll see that
```
red =>	1.045830249786377
green =>	0.73694425821304321
blue	=> -0.059313245117664337
```

So, I think that there is some code that is clamping the values OR the interpolation formula does not work properly.

<img width="998" alt="Screenshot 2024-01-08 at 15 40 11" src="https://github.com/ryanlntn/react-native/assets/11162307/eb8b3e71-8dbd-416e-8f2b-9cd622ff77d7">

The other problem is that [`RCTFromColorComponent`](https://github.com/facebook/react-native/blob/782e9eace974baefba7d8978739f11fe1ea70dc9/packages/react-native/Libraries/NativeAnimation/RCTAnimationUtils.mm#L105) is not aware of the color space... So we need to update `RCTInterpolateColorInRange` to take the color space into consideration: if a displayP3 color is passed as input, it should provide a displayP3 color as output.
I suspect that, as of now, it is returning a  sRGB color.